src/ViewModels/src/DeploymentDialog/DeploymentDialogViewModel.cs

```diff
@@ -841,10 +842,15 @@ public void RemoveFromSelectedServices(object arg)
 
         public void ClearSelectedServices(object arg = null)
         {
-            SelectedServices.Clear();
-            foreach (var svItem in ServiceOptions)
+            var initialSelection = SelectedServices.ToList(); // copy to avoid iterating over a collection that's being modified
```

Comment:
If the only purpose of this var is as comment suggests, you can write it inline as such:
```
            foreach (var svcName in SelectedServices.ToList())
```